### PR TITLE
fix for rfa 01b disclosures bug cals-5477

### PR DESCRIPTION
**Pull request by lauherkcambria**
#### Which User story does this relate to?
-https://osi-cwds.atlassian.net/browse/CALS-5477

#### Brief feature description
- provides a fix for entering a criminal record disclosure in the RFA 01B form. 

#### Does this PR block/effect other stories?
-no

#### Do any commands needs to be run in order to run the project with this PR?
-no

#### Code coverage percent diff as a result of this PR?
-

### CODE REVIEW

**Comment by lauherkcambria** [comment]
jenkins, retest this please.

**Comment by Nasir2712** [comment]
@lauherkcambria the story attached to the description box redirects to "Cannot enter text in RFA 01C text boxes for relationships" bug. Can you link the right story please?

**Comment by Nasir2712** [comment]
@lauherkcambria can you add feature tests for this card?

**Comment by ratneshraval** [comment]
Can you put story number in PR name?

File: app/javascript/rfa_forms/rfa01b_edit_view/index.js [hunk 1]
```diff
@@ -180,7 +180,7 @@ export default class Rfa01bList extends React.Component {
             <CardsGroupLayout>
               <CrimeBackgroundAgainstCohabitant
                 arrestedForCrime={this.state.application.arrested_for_crime}
-                disclosures={this.state.application.arrested_for_crime_disclosures}
+                disclosures={checkArrayObjectPresence(this.state.application.arrested_for_crime_disclosures) || [disclosureDefaults]}
```
Comment (ratneshraval) [inline app/javascript/rfa_forms/rfa01b_edit_view/index.js]: So why are we still doing `checkArrayObjectPresence`? This used to be not an issue on this form, right? Let's have this fixed on the API.

**Comment by ratneshraval** [comment]
I also noticed that, disclosureDefaults are now in constant, californiaCriminalBackground.js and californiaCriminalBackground_test.js. Should consolidate that to constants only and use from there.

File: spec/features/rfa_01b_application_spec.rb [hunk 2]
```diff
@@ -4,7 +4,7 @@
 
 RSpec.feature 'RFA01B', js: true do
 
-  scenario 'validate rfa01b ', set_auth_header: true do
+  scenario 'validate rfa01b', set_auth_header: true do
```
[the remark this answers is not in the document]
Comment (lauherkcambria) [reply spec/features/rfa_01b_application_spec.rb]: good idea

[the remark this answers is not in the document]
Comment (lauherkcambria) [reply spec/features/rfa_01b_application_spec.rb]: @ratneshraval, added scenarios for all other disclosure types 😄

File: spec/features/rfa_01b_application_spec.rb [hunk 3]
```diff
@@ -27,24 +27,43 @@
...
-    #find(:select, 'offense_state').first(:option, 'Alaska').select_option
-    #fill_in('offense_date', with: 'date of offense', :match => :prefer_exact)
-    #fill_in('offense_details', with:'offense details' , :match => :prefer_exact)
+    find('[for=californiaCriminalBackgroundRadiotrue]').click
+    fill_in('californiaCriminalBackgroundoffenseReason', with: 'test offense', :match => :prefer_exact)
+    fill_in('californiaCriminalBackgroundoffenseCity', with: 'test offense city' , :match => :prefer_exact)
+    fill_in('californiaCriminalBackgroundOffenseDate', with: 'date of offense', :match => :prefer_exact)
+    fill_in('californiaCriminalBackgroundoffenseDetails', with:'offense details' , :match => :prefer_exact)
+
+    find('[for=crimeBackgroundAgainstCohabitantRadiotrue').click
```
Comment (Nasir2712) [inline spec/features/rfa_01b_application_spec.rb]: we can add scenarios block so that it'll be helpful for error tracking incase if we get one.

**Pull request merged**